test/workers/delete_object_hierarchy_worker_test.rb

```diff
@@ -122,7 +122,7 @@ class ObjectCachingTest < ActiveSupport::TestCase
       # cms_template is obtained by ID once then reused for each association; and once reloaded before being destroyed
       assert_number_of_queries(2, matching: /SELECT.*\bcms_templates\b.*\bid['"` ]{0,2}=/) do
         # cms_templates_versions are never loaded by id as they are just `take`n, cached and not reloaded before destroy
-        assert_number_of_queries(0, matching: /SELECT.*\bcms_templates_version\b.*\bid['"` ]{0,2}=/) do
+        assert_number_of_queries(0, matching: /SELECT.*\bcms_templates_versions\b.*\bid['"` ]{0,2}=/) do
```

Comment:
that was a bug from the previous PR, in fact it turned out we were not caching the object upon retrieval from association, but now we do